src/lib/statistics/statistics_objects/equal_distinct_count_histogram.cpp

```diff
@@ -49,11 +50,36 @@ template <typename T>
 std::vector<std::pair<T, HistogramCountType>> value_distribution_from_column(const Table& table,
                                                                              const ColumnID column_id,
                                                                              const HistogramDomain<T>& domain) {
-  // TODO(anybody) If you want to look into performance, this would probably benefit greatly from monotonic buffer
-  //               resources.
-  ValueDistributionMap<T> value_distribution_map;
-
+  // Since this function can be the main bottleneck for loading data in the benchmarks, we put some effort in
+  // optimizing it. For once, we try to get the distinct count in case we have dictionary-encoded data in place. This
+  // allows us to pre-size the robin_map which helps a lot with performance (no resizes for large maps) and memory
+  // consumption (as we can avoid pre-sizes all maps with the table size, which is often by far too large).
+
+  // Use a defensive default in case we cannot get a better estimate (i.e., no dictionary-encoded data).
+  auto estimated_distinct_value_count = std::min(table.row_count(), uint64_t{1'024});
+  auto max_load_factor = std::optional<float>{};
   const auto chunk_count = table.chunk_count();
+
+  // Meta tables do not have chunks, we thus need to check for having a chunk count larger zero.
+  if (chunk_count > 0 && table.get_chunk(ChunkID{0})) {
```

Comment:
Why do we need both checks? Wouldn't the first one be sufficient for ruling out a meta table?

Reply:
A previously existing chunk could have been removed (`Table::remove_chunk`), i.e, the pointer has been set to `nullptr`.